my-react-app/src/components/GetData/DataItem.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from "react";
+import clsx from "clsx";
+import Modal from "react-modal";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTimes } from "@fortawesome/free-solid-svg-icons";
+
+Modal.setAppElement("#root");
+
+type Data = {
+  id: number;
+  title: string;
+  year: string;
+  genre: string;
+  director: string;
+  actors: string;
+  language: string;
+  production: string;
+  website: string;
+};
+
+type Props = {
+  data: Data;
+};
+
+const DataItem = ({ data }: Props) => {
+  const [modalIsOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <button
+        onClick={() => setIsOpen(true)}
+        key={data.id}
+        className="p-8 border-2 rounded-lg hover:bg-zinc-200 duration-15 basis-full md:basis-2/4 lg:basis-1/4 hover:scale-110"
+      >
+        <h1 className="font-bold">{data.title}</h1>
+      </button>
+      <Modal
```

Comment:
versuche das modal in eine extra componentente + file auszulagern und die benötigten daten über die props rein zu geben